frontend/packages/console-dynamic-plugin-sdk/src/extensions/custom.ts

```diff
@@ -0,0 +1,8 @@
+import { ExtensionDeclaration } from '../types';
+
+/**
+ * Custom Console extension with arbitrary `type` and `properties`.
+ *
+ * This type represents all non-standard Console extension declarations.
```

Comment:
Your PR doesn't describe what this is solving or link out to a ticket. Can you help me understand why we need this?

Reply:
I'd mention this is so plugins can define their own extension points for other plugins to use.

Reply:
Done, the `CustomExtension` type doc should be better now :smiley: